### ISSUE TITLE
hermit-abi 0.3.2 git tag and yanking explanation

### ISSUE THREAD

**Issue by faern** [hermitcore/rusty-hermit#436]
Hi!

This crate showed up in our `cargo audit` CI job today since we were using `0.3.1` which was now yanked. My usual process then is to check why it was yanked and see what upgrades are available. To know how critical it is to upgrade the dependency. However, finding that information was not super easy. I finally found it in the commit message of the latest commit in `master`. This issue is a friendly reminder/request to:

* Push a tag for `hermit-abi-0.3.2` - I'm sure this is just forgotten, because you are good at pushing tags otherwise it seems.
* Introduce a changelog - Makes it way easier for a library consumer to understand what changed between versions (including why a version was yanked).

**Comment by mkroening** [comment]
This is the only change in that version: https://github.com/hermitcore/rusty-hermit/commit/ca10ddea80c418a77e4213d9212d77079e56f094.

Regarding tags: Since we have two crates which are published at different times, I found tags to be confusing. Still, it might make sense to still have them rather than not. 👍

**Comment by faern** [comment]
Having no tags makes it very hard to match the code on crates.io with the code in the git repository. I personally find having tags (preferably signed) to be really good and helpful in many ways. Especially around supply chain security.

**Comment by asomers** [comment]
IMHO there's no need to yank a release for a minor bug like that.  Yanking should be reserved for things like security vulnerabilities, or backwards-compatibility violations.

**Comment by mkroening** [comment]
We have created tags for the `hermit-abi` versions. :)

> IMHO there's no need to yank a release for a minor bug like that.

For Hermit, this was a huge backwards compatibility violation. Versions 0.3.0 and 0.3.1 with the typo tried to link against a non-existing symbol. This only showed up when version 0.3 was adapted by `num-cpus`. I think, this justifies yanking as described in [The Cargo Book](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-yank), don't you think?

Closing this. You are welcome to reopen if you have objections. :)
